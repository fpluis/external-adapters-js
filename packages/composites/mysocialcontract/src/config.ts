import { Requester } from '@chainlink/ea-bootstrap'
import { Config } from '@chainlink/types'

export const NAME = 'MYSOCIALCONTRACT'

export const DEFAULT_ENDPOINT = 'statistics'
export const DEFAULT_BASE_URL = 'https://youtube.googleapis.com'

export const makeConfig = (prefix?: string): Config => {
  const config = Requester.getDefaultConfig(prefix)
  config.api.baseURL = config.api.baseURL || DEFAULT_BASE_URL
  config.defaultEndpoint = DEFAULT_ENDPOINT
  return config
}
