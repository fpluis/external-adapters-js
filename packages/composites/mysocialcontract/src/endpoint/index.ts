export * as socials from './socials'
