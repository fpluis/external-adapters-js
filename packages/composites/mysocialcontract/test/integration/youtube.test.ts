import { AdapterRequest } from '@chainlink/types'
import request from 'supertest'
import * as process from 'process'
import { server as startServer } from '../../src'
import * as nock from 'nock'
import * as http from 'http'
import { mockResponseSuccess } from './fixtures'

describe('execute', () => {
  const id = '1'
  let server: http.Server
  const req = request('localhost:8080')
  beforeAll(async () => {
    process.env.CACHE_ENABLED = 'false'
    process.env.YOUTUBE_API_KEY = 'mock-api-key'
    if (process.env.RECORD) {
      nock.recorder.rec()
    }
    server = await startServer()
  })
  afterAll((done) => {
    if (process.env.RECORD) {
      nock.recorder.play()
    }

    nock.restore()
    nock.cleanAll()
    nock.enableNetConnect()
    server.close(done)
  })

  describe('statistics api', () => {
    it('should return success with only one yt property', async () => {
      const data: AdapterRequest = {
        id,
        data: {
          ytChannelId: 'UCfpnY5NnBl-8L7SvICuYkYQ',
          ytViews: true,
        },
      }
      mockResponseSuccess()

      const response = await req
        .post('/')
        .send(data)
        .set('Accept', '*/*')
        .set('Content-Type', 'application/json')
        .expect('Content-Type', /json/)
        .expect(200)
      expect(response.body).toEqual({
        data: {
          result: {
            ytViews: 30692800,
          },
        },
        jobRunID: '1',
        result: {
          ytViews: 30692800,
        },
        statusCode: 200,
      })
    })

    it('should return success with both yt properties', async () => {
      const data: AdapterRequest = {
        id,
        data: {
          ytChannelId: 'UCfpnY5NnBl-8L7SvICuYkYQ',
          ytViews: true,
          ytSubs: true,
        },
      }
      mockResponseSuccess()

      const response = await req
        .post('/')
        .send(data)
        .set('Accept', '*/*')
        .set('Content-Type', 'application/json')
        .expect('Content-Type', /json/)
        .expect(200)
      expect(response.body).toEqual({
        data: {
          result: {
            ytViews: 30692800,
            ytSubs: 104000,
          },
        },
        jobRunID: '1',
        result: {
          ytViews: 30692800,
          ytSubs: 104000,
        },
        statusCode: 200,
      })
    })
  })
})
