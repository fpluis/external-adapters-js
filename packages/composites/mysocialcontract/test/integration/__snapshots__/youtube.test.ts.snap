// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute statistics api should return success 1`] = `
Object {
  "data": Object {
    "result": Object {
      "hiddenSubscriberCount": false,
      "subscriberCount": "104000",
      "videoCount": "1551",
      "viewCount": "30692800",
    },
  },
  "jobRunID": "1",
  "result": Object {
    "hiddenSubscriberCount": false,
    "subscriberCount": "104000",
    "videoCount": "1551",
    "viewCount": "30692800",
  },
  "statusCode": 200,
}
`;
